hudi-spark/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -65,15 +65,20 @@ class IncrementalRelation(val sqlContext: SQLContext,
       s"option ${DataSourceReadOptions.BEGIN_INSTANTTIME_OPT_KEY}")
   }
 
-  val lastInstant = commitTimeline.lastInstant().get()
+  private val lastInstant = commitTimeline.getReverseOrderedInstants.iterator().find(instant => {
+    // Skip empty instants
+    HoodieCommitMetadata
+      .fromBytes(commitTimeline.getInstantDetails(instant).get(), classOf[HoodieCommitMetadata])
+      .fetchTotalRecordsWritten() > 0
+  }).getOrElse(throw new HoodieException("Couldn't find any non empty instant in the hoodie commit timeline."))
 
-  val commitsToReturn = commitTimeline.findInstantsInRange(
+  private val commitsToReturn = commitTimeline.findInstantsInRange(
     optParams(DataSourceReadOptions.BEGIN_INSTANTTIME_OPT_KEY),
     optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME_OPT_KEY, lastInstant.getTimestamp))
     .getInstants.iterator().toList
 
   // use schema from a file produced in the latest instant
-  val latestSchema = {
+  private val latestSchema = {
```

Comment:
Original Avro Schema (without hudi fields) is now getting written in commit metadata. Can we try reading it first and fallback to reading from parquet if we cannot find schema in commit metadata ?

Reply:
Done, fallback to reading from parquet was already handled by `TableSchemaResolver`